### DISCUSSION TITLE
How does MPD handle ALSA?

### DISCUSSION THREAD

**Discussion by LarsGrootkarzijn** [MusicPlayerDaemon/MPD#1792] [Q&A] [answered]
Hello all,

I have a custom ALSA-lib for my device to handle my soundcard correctly. Now, this lib does not work with MPD at the moment. When playbacking music via MPD how does this get handled? It seems like MPD never sends a command to alsa-lib to change the bitrate. Usually there would be a call to `snd_pcm_hw_param_set()` to change the bitrate. For example when 44.1khz this function would get called with 44100. However, MPD does not seem to do this. Is this correct?

**Comment by MaxKellermann** [comment, marked as answer]
You mean sample rate, not bit rate?

> Usually there would be a call to `snd_pcm_hw_param_set()` to change the bitrate.

[quotation needed]

And here's my quotation right from the MPD source code:
https://github.com/MusicPlayerDaemon/MPD/blob/99885c4cbcbb5545708104825cb56d67e20c4517/src/lib/alsa/HwSetup.cxx#L203-L204

This call has been working fine for MPD for the last 20 years.

**Reply by LarsGrootkarzijn** [reply]
Hi, I indeed did mean sample rate.
Thank you for you answer, I will probably change alsalib a bit more to work with `snd_pcm_hw_params_set_rate_near` aswell. Thank you.
